test/controllers/changePref/EmailToUseControllerSpec.scala

```diff
@@ -0,0 +1,179 @@
+import connectors.httpParsers.GetCustomerInfoHttpParser.GetCustomerInfoResponse
+import controllers.ControllerBaseSpec
+import forms.YesNoForm.yesNo
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.changePref.EmailToUseView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EmailToUseControllerSpec extends ControllerBaseSpec {
+
+  def setup(result: GetCustomerInfoResponse): Any =
+    when(mockVatSubscriptionService.getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext]))
+      .thenReturn(Future.successful(result))
+
+  val testValidationEmail: String = "validation@example.com"
+
+  lazy val existingEmailSessionRequest: FakeRequest[AnyContentAsEmpty.type] =
+    request.withSession(SessionKeys.validationEmailKey -> testValidationEmail)
+
+  lazy val noEmailSessionRequest: FakeRequest[AnyContentAsEmpty.type] =
+    request
+
+  val view: EmailToUseView = injector.instanceOf[EmailToUseView]
+
+  def target(result: GetCustomerInfoResponse = Right(fullCustomerInfoModel)): EmailToUseController = {
+    setup(result)
+
+    new EmailToUseController(
+      mockVatSubscriptionService,
+      mockErrorHandler,
+      view
+    )
+  }
+
+  "Calling the show action in EmailToUseController" when {
+
+    "the letterToConfirmedEmail switch is enabled" when {
+
+      "there is an email in session" should {
+
+        lazy val result = {
+          mockConfig.features.letterToConfirmedEmailEnabled(true)
+          mockIndividualAuthorised()
+          target().show()(existingEmailSessionRequest)
+        }
+
+        "return 200 (OK)" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+      }
+
+      "there isn't an email in session" should {
+
+        lazy val result = {
+          mockConfig.features.letterToConfirmedEmailEnabled(true)
+          mockIndividualAuthorised()
+          target().show()(noEmailSessionRequest)
+        }
+
+        "return 200 (OK)" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+      }
+    }
+
+    "the letterToConfirmedEmail switch is disabled" should {
+
+      "return a 400" in {
+
+        lazy val result = {
+          mockConfig.features.letterToConfirmedEmailEnabled(false)
+          mockIndividualAuthorised()
+          target().show()(existingEmailSessionRequest)
+        }
+        status(result) shouldBe Status.BAD_REQUEST
+      }
+    }
+  }
+
+  "Calling the submit action in EmailToUseController" when {
```

Comment:
Could you also add tests for the two code branches:

- User has no email in session
- User submits the form without selecting any option